### PR TITLE
EVG-3125: move idle host flagging out of the monitor

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: units/crons.go [hunk 1]
```diff
@@ -248,6 +248,37 @@ func PopulateHostTerminationJobs(env evergreen.Environment) amboy.QueueOperation
...
+
+		catcher := grip.NewBasicCatcher()
+		ts := util.RoundPartOfHour(1).Format(tsFormat)
+		hosts, err := host.AllIdleEphemeral()
+		if err != nil {
+			return errors.Wrap(err, "problem finding idle hosts")
+		}
+
+		for _, h := range hosts {
+			catcher.Add(queue.Put(NewIdleHostTerminationJob(env, h, ts)))
```
Comment (john-m-liu) [inline units/crons.go]: Would it be better to move the idle time checking here, and only add a host termination job if we want to terminate it? That way we won't add a bunch of no-op jobs to the queue

File: units/host_monitoring_idle_termination.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewIdleHostTerminationJob(env evergreen.Environment, h host.Host, id string) amboy.Job {
+	j := makeIdleHostJob()
+	j.host = &h
```
Comment (john-m-liu) [inline units/host_monitoring_idle_termination.go]: Does this do anything if the host doesn't get persisted to the db?

File: units/crons.go (hunk 1 above)
Comment (tychoish) [reply units/crons.go]: for distros with hourly billing, we have to do aws queries and interact with the host provider. We could cache more of this information in the database and then add a clause to the `host.FindHostsToTerminate()` query, but for now I think we need to do some of this this logic application side.

noop jobs aren't really a problem?

File: units/host_monitoring_idle_termination.go (hunk 2 above)
Comment (tychoish) [reply units/host_monitoring_idle_termination.go]: it only does things if the job doesn't get persisted to the db.

**Pull request merged**